core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -735,7 +749,12 @@ object SparkSubmit extends CommandLineUtils {
     }
 
     try {
-      mainMethod.invoke(null, childArgs.toArray)
+      if (isSparkApp) {
+        val envvars = Map[String, String]() ++ sys.env
+        mainMethod.invoke(null, childArgs.toArray, childSparkConf, envvars.toMap)
```

Comment:
You're just passing `sys.env` as the environment map to the SparkApp. I've already mentioned this before, multiple times. If you're not going to modify `sys.env`, then just remove the argument.

(And adding an empty map to `sys.env` does not count as modifying it.)